cmd/containerd-shim-runhcs-v1/task.go

```diff
@@ -95,6 +98,30 @@ type shimTask interface {
 	// If the host is hypervisor isolated and this task owns the host additional
 	// metrics on the UVM may be returned as well.
 	Stats(ctx context.Context) (*stats.Statistics, error)
+	// Update updates a task's container
+	Update(ctx context.Context, req *task.UpdateTaskRequest) error
+}
+
+func verifyTaskUpdateResourcesType(data interface{}) error {
+	switch data.(type) {
+	case *specs.WindowsResources:
+	case *specs.LinuxResources:
+	default:
+		return errors.New("update resources must be of type *WindowsResources or *LinuxResources")
+	}
+	return nil
+}
+
+func normalizeContainerMemorySize(ctx context.Context, cid string, requestedSize uint64) uint64 {
+	actual := (requestedSize + 1) &^ 1 // align up to an even number
+	if requestedSize != actual {
+		log.G(ctx).WithFields(logrus.Fields{
+			"containerID": cid,
+			"requested":   requestedSize,
+			"assigned":    actual,
+		}).Warn("Changing user requested MemorySizeInMB to align to 2MB")
+	}
+	return actual
```

Comment:
We have a function that does the exact same thing in `uvm/create.go` can we just move that to a common file somewhere and use that at both locations?

Reply:
yeah I saw that. I'm not sure what a good common place would be though. Any suggestions?

Reply:
I think we can create a new file at `internal/helpers.go` and put this in there? Maybe a few other commons functions can go in that file too?

Reply:
Did we decide something about this?